src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -20,10 +21,10 @@ public static void CopyFile(string sourceFullPath, string destFullPath, bool ove
             }
 
             // Copy the contents of the file from the source to the destination, creating the destination in the process
-            using (var src = new FileStream(sourceFullPath, FileMode.Open, FileAccess.Read, FileShare.Read, DefaultBufferSize, FileOptions.None))
-            using (var dst = new FileStream(destFullPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None, DefaultBufferSize, FileOptions.None))
+            using (SafeFileHandle src = File.OpenHandle(sourceFullPath, FileMode.Open, FileAccess.Read, FileShare.Read, FileOptions.SequentialScan))
```

Comment:
I wonder if `FileOptions.SequentialScan` is actually going to help as it requires an extra syscall on Unix:

https://github.com/dotnet/runtime/blob/d841f5e17897be5ef6aef5031b8a03ed3d8fd8dd/src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs#L278-L284

would be nice to benchmark it for small, medium and large files

Reply:
I'll remove it to unblock this PR.  But if we don't believe it's helpful on Unix, the right action isn't to avoid using it in cross-platform code but rather to stop issuing the PosixFAdvise call for the hint.